src/test/java/dev/persiantools/BankTest.java

```diff
@@ -0,0 +1,145 @@
+package dev.persiantools;
+
+import dev.persiantools.usecases.exceptions.BankNotFoundByProvidedCardNumber;
+import dev.persiantools.usecases.exceptions.BankNotFoundByProvidedIban;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+public class BankTest
+{
+    @Test
+    public void shouldReturnsBanksNameUsingCardNumbers()
+    {
+        BankUtils bank = new BankUtils();
+
+        try {
+            for(Map.Entry<String, String> entry : getBanksDetailsByIban().entrySet()) {
+                String iban = entry.getKey();
+                String name = entry.getValue();
+
+                assertEquals(bank.findByIban(iban).getPersianName(), name);
```

Comment:
For a consistent test it is better to use another parameter rather than `persianName` because Persian characters are too tricky for equality checks.

Reply:
Correct.

Reply:
It sounds good